### PR TITLE
Add support for sendCOMM method

### PR DESCRIPTION
**Pull request by sanderdatema**
Credits go to user angelsware, I just did a copy/paste from #120

### CODE REVIEW

**Comment by sanderdatema** [comment]
The build process tries to use the MacOS 10.14 DSK, but that's not installed (`xcodebuild: error: SDK "macosx10.14" cannot be located.`). Is that a fix I should do or do you need to fix it on your side?

File: Java/src/XPlaneConnect.java [hunk 1]
```diff
@@ -173,32 +185,27 @@ public XPlaneConnect(String xpHost, int xpPort, int port, int timeout)
...
+    private byte[] readUDP() throws IOException {
+        final byte[] buffer = new byte[65536];
+        final DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+        try {
             socket.receive(packet);
             return Arrays.copyOf(buffer, packet.getLength());
-        }
-        catch (SocketTimeoutException ex)
-        {
+        } catch (final SocketTimeoutException ex) {
```
Comment (jason-watkins) [inline Java/src/XPlaneConnect.java:208]: Most of these formatting changes I'm okay with, but I absolutely cannot stand `catch` (or `else`/`else if`) being on the same line as the previous closing brace

File: Java/src/XPlaneConnect.java [hunk 2]
```diff
@@ -221,110 +227,100 @@ private void sendUDP(byte[] buffer) throws IOException
      * @param pause {@code true} to pause the simulator; {@code false} to un-pause.
      * @throws IOException If the command cannot be sent.
      */
-    public void pauseSim(boolean pause) throws IOException
-    {
-        //            S     I     M     U     LEN   VAL
-        byte[] msg = {0x53, 0x49, 0x4D, 0x55, 0x00, 0x00};
-        msg[5] = (byte)(pause ? 0x01 : 0x00);
+    public void pauseSim(final boolean pause) throws IOException {
+        // S I M U LEN VAL
```
Comment (jason-watkins) [inline Java/src/XPlaneConnect.java:231]: The white space in this comment was important.

File: Java/src/XPlaneConnect.java [hunk 3]
```diff
@@ -221,110 +227,100 @@ private void sendUDP(byte[] buffer) throws IOException
...
+     */
+    public void pauseSim(final int pause) throws IOException {
+        if (pause < 0 || (pause > 2 && pause < 100) || (pause > 119 && pause < 200) || pause > 219) {
             throw new IllegalArgumentException("pause must be a value in the range [0, 2], [100, 119], or [200, 219].");
         }
 
-        //            S     I     M     U     LEN   VAL
-        byte[] msg = {0x53, 0x49, 0x4D, 0x55, 0x00, 0x00};
-        msg[5] = (byte)pause;
+        // S I M U LEN VAL
```
Comment (jason-watkins) [inline Java/src/XPlaneConnect.java:251]: Significant white space again

File: Java/src/XPlaneConnect.java [hunk 4]
```diff
@@ -457,102 +444,98 @@ public void sendDREFs(String[] drefs, float[][] values) throws IOException
...
+                if (i >= values.length) {
+                    bb.put(cur, (byte) -1);
+                } else {
+                    bb.put(cur, (byte) values[i]);
                 }
                 cur += 1;
-            }
-            else if (i >= values.length)
-            {
+            } else if (i >= values.length) {
```
Comment (jason-watkins) [inline Java/src/XPlaneConnect.java:524]: See comment above. move `else if` to the next line.

File: Java/src/XPlaneConnect.java [hunk 5]
```diff
@@ -457,102 +444,98 @@ public void sendDREFs(String[] drefs, float[][] values) throws IOException
...
                 bb.putFloat(cur, -998);
-                cur+= 4;
-            }
-            else
-            {
+                cur += 4;
+            } else {
```
Comment (jason-watkins) [inline Java/src/XPlaneConnect.java:527]: This one too

File: Java/src/XPlaneConnect.java [hunk 6]
```diff
@@ -598,89 +579,88 @@ else if (i >= values.length)
...
-            if(i<3) /* lat/lon/height as double */
+        for (i = 0; i < values.length; ++i) {
+            if (i < 3) /* lat/lon/height as double */
             {
                 bb.putDouble(values[i]);
-            }
-            else
-            {
-                bb.putFloat((float)values[i]);
+            } else {
```
Comment (jason-watkins) [inline Java/src/XPlaneConnect.java:652]: This one too

File: Python/src/xpc.py [hunk 7]
```diff
@@ -1,20 +1,22 @@
...
 class XPlaneConnect(object):
-    '''XPlaneConnect (XPC) facilitates communication to and from the XPCPlugin.'''
+    """XPlaneConnect (XPC) facilitates communication to and from the XPCPlugin."""
+
     socket = None
 
     # Basic Functions
-    def __init__(self, xpHost = 'localhost', xpPort = 49009, port = 0, timeout = 100):
-        '''Sets up a new connection to an X-Plane Connect plugin running in X-Plane.
+    def __init__(self, xpHost="localhost", xpPort=49009, port=0, timeout=100):
```
Comment (jason-watkins) [inline Python/src/xpc.py:11]: Is there any specific reason to change the single quotes to double? If not, I'd really prefer to just leave things as they are.

I really like the white space around my assignment operators as well.

**Comment by sanderdatema** [comment]
I was hoping reverting the changes of e1a3478 would help. But nope.

File: Python/src/xpc.py [hunk 8]
```diff
@@ -24,7 +26,9 @@ def __init__(self, xpHost = 'localhost', xpPort = 49009, port = 0, timeout = 100
             raise ValueError("Unable to resolve xpHost.")
 
         if xpPort < 0 or xpPort > 65535:
-            raise ValueError("The specified X-Plane port is not a valid port number.")
+            raise ValueError(
```
Comment (jason-watkins) [inline Python/src/xpc.py:29]: Wrapping at 80 characters is just too aggressive for modern environments. No hard-wrapping at less than 120 characters please.

File: xpcPlugin/DataManager.h [hunk 9]
```diff
@@ -410,6 +410,13 @@ namespace XPC
 		/// \param value The flaps settings. Should be between 0.0 (no flaps) and 1.0 (full flaps).
 		static void SetFlaps(float value);
 
+        /// Executes a command
+        ///
+        /// \param comm   The name of the command to execute.
+        static void Execute(const std::string& comm);
+
+        /// Gets a default value that indicates that a dataref should not be changed.
```
Comment (jason-watkins) [inline xpcPlugin/DataManager.h:418]: I think this was included accidentally?

Also, the indentation looks off here. That usually means there's mixed tabs and spaces.

**Comment by jason-watkins** [comment]
I'll look into the build failure a little later. In the meantime, there are a few formatting issues that  I'd like to resolve before I merge.

Thanks for be proactive on the CLA!

File: Java/src/XPlaneConnect.java (hunk 1 above)
Comment (sanderdatema) [reply Java/src/XPlaneConnect.java:208]: Fair enough. These were done automatically by Black, following PEP rules. But I'd rather have this fixed soon than getting stuck in cosmetic things. 😀

File: xpcPlugin/DataManager.h (hunk 9 above)
Comment (sanderdatema) [reply xpcPlugin/DataManager.h:418]: That's a weird one. I'll look into that.

Comment (sanderdatema) [reply xpcPlugin/DataManager.h:418]: Actually this was in the original pull request: https://github.com/nasa/XPlaneConnect/pull/120/commits/9bfa8c3f24eb1156883f1fd19e5f6c24f7e632c8#diff-c14463a65b9022a134315f9ddc438ec4

**Comment by jason-watkins** [comment]
Ah, sorry for not noticing this earlier... The PR should be branched from and should target the `develop` branch. I believe this will also fix the macOS failure on Azure, as the develop branch is currently targeted at 10.15 rather than 10.14 on master.

**Comment by sanderdatema** [comment]
Ok... Well, I'll start again then. :)

**Pull request closed without merging**